apps/gitness/src/utils/git-utils.ts

```diff
@@ -1,5 +1,131 @@
+import langMap from 'lang-map'
+
 export const REFS_TAGS_PREFIX = 'refs/tags/'
 export const REFS_BRANCH_PREFIX = 'refs/heads/'
+const MONACO_SUPPORTED_LANGUAGES = [
+  'abap',
+  'apex',
+  'azcli',
+  'bat',
+  'bicep',
+  'cameligo',
+  'clojure',
+  'coffee',
+  'cpp',
+  'csharp',
+  'csp',
+  'css',
+  'cypher',
+  'dart',
+  'dockerfile',
+  'ecl',
+  'elixir',
+  'flow9',
+  'freemarker2',
+  'fsharp',
+  'go',
+  'graphql',
+  'handlebars',
+  'hcl',
+  'html',
+  'ini',
+  'java',
+  'javascript',
+  'json',
+  'julia',
+  'kotlin',
+  'less',
+  'lexon',
+  'liquid',
+  'lua',
+  'm3',
+  'markdown',
+  'mips',
+  'msdax',
+  'mysql',
+  'objective-c',
+  'pascal',
+  'pascaligo',
+  'perl',
+  'pgsql',
+  'php',
+  'pla',
+  'postiats',
+  'powerquery',
+  'powershell',
+  'protobuf',
+  'pug',
+  'python',
+  'qsharp',
+  'r',
+  'razor',
+  'redis',
+  'redshift',
+  'restructuredtext',
+  'ruby',
+  'rust',
+  'sb',
+  'scala',
+  'scheme',
+  'scss',
+  'shell',
+  'solidity',
+  'sophia',
+  'sparql',
+  'sql',
+  'st',
+  'swift',
+  'systemverilog',
+  'tcl',
+  'twig',
+  'typescript',
+  'vb',
+  'wgsl',
+  'xml',
+  'yaml'
+]
+
+export const PLAIN_TEXT = 'plaintext'
+
+// Some of the below languages are mapped to Monaco's built-in supported languages
+// due to their similarity. We'll still need to get native support for them at
+// some point.
+const EXTENSION_TO_LANG: Record<string, string> = {
+  alpine: 'dockerfile',
+  bazel: 'python',
+  cc: 'cpp',
+  cs: 'csharp',
+  env: 'shell',
+  gitignore: 'shell',
+  jsx: 'typescript',
+  makefile: 'shell',
+  toml: 'ini',
+  tsx: 'typescript',
+  tf: 'hcl',
+  tfvars: 'hcl',
+  workspace: 'python',
+  tfstate: 'hcl',
+  ipynb: 'json'
+}
+
+/**
+ * Convert number of bytes into human readable format
+ *
+ * @param integer bytes     Number of bytes to convert
+ * @param integer precision Number of digits after the decimal separator
+ * @return string
+ * @link https://stackoverflow.com/a/18650828/1114931
+ */
+export function formatBytes(bytes: number, decimals = 2) {
+  if (!+bytes) return '0 Bytes'
+
+  const k = 1024
+  const dm = decimals < 0 ? 0 : decimals
+  const sizes = ['Bytes', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB']
```

Comment:
NIT: do we need to show as KiB or KB format? Which one is more universal?

Reply:
Agree, probably just use `KB` format.

Reply:
sure